apps/public/example_test.go

```diff
@@ -0,0 +1,46 @@
+package public_test
+
+import (
+	"context"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/public"
+)
+
+// This example demonstrates the general pattern for authenticating with MSAL Go:
+//   - create a client (only necessary at application start--it's best to reuse client instances)
+//   - call AcquireTokenSilent() to search for a cached access token
+//   - if the cache misses, acquire a new token
+func Example() {
+	client, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/your_tenant"))
+	if err != nil {
+		// TODO: handle error
+	}
+
+	var result public.AuthResult
+	scopes := []string{"scope"}
+
+	// If your application previously authenticated a user, call AcquireTokenSilent with that user's account
+	// to use cached authentication data. This example shows choosing an account from the cache, however this
+	// isn't always necessary because the AuthResult returned by authentication methods includes user account
+	// information.
+	accounts, err := client.Accounts(context.TODO())
+	if err != nil {
+		// TODO: handle error
+	}
+	if len(accounts) > 0 {
+		// There may be more accounts; here we assume the first one is wanted.
+		// AcquireTokenSilent returns a non-nil error when it can't provide a token.
+		result, err = client.AcquireTokenSilent(context.TODO(), scopes, public.WithSilentAccount(accounts[0]))
+	}
+	if err != nil || len(accounts) == 0 {
```

Comment:
Thanks, @chlowell, for engaging in the recent discussion and making corresponding changes!

Nit: Was this line of now-complicated logic part of the reason that you did not want to go this route? Perhaps we can simply it like this.
```suggestion
	if result == nil {
```

Besides, the code snippet in README seems not as polished as this examples_test.go file.

Those issues are minor. I'll leave it to you to decide whether further polish it. Meanwhile, I'm approving this PR.

Reply:
Yes, I don't like conditionally calling `AcquireTokenSilent` because it makes the silent auth failure case more awkward. It would be nice to examine `result` instead of `err` and `accounts` (again!) but `result` can't be compared to `nil` because it isn't a pointer. It is zero-valued when either condition is true, however we'd have to reflect to establish that.

Thanks for reminding me to copy this code to the readme 😊